cmd/_integration-tests/transport/http_test.go

```diff
@@ -400,6 +404,124 @@ func TestStrangeRPCName(t *testing.T) {
+func TestNonJSONResponseBodyFromClientCallIsLessThan8KB(t *testing.T) {
+	svchttp, err := httpclient.New(nonJSONHTTPAddr)
+	if err != nil {
+		t.Fatalf("cannot create httpclient: %q", err)
+	}
+
+	var req pb.Empty
+	_, err = svchttp.ErrorRPCNonJSONLong(context.Background(), &req)
+	if err == nil {
+		t.Fatal("Expected error from non-json response with http client")
+	}
+
+	l := len(err.Error())
+	// Add 200 for padding for the actual error message in addition to the response body
+	if l > 8196+200 {
+		t.Fatalf("Expected error to be less than 8KB with a little padding, actual %d", l)
+	}
+	t.Log("Non JSON response length", l)
+}
+
+// Test that if a truss server receives a non-json request, we put that request
+// body in the error message. To allow for developers to see the request body in the errors.
+func TestNonJSONRequestBodyIsInError(t *testing.T) {
+	// Put some bad data into the body
+	req, err := http.NewRequest("GET", httpAddr+"/error/non/json", strings.NewReader(brokenHTTPRequest))
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "cannot construct http request"))
+	}
+
+	respBytes, err := testHTTPRequest(req)
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "cannot make http request"))
+	}
+
+	var resp map[string]string
+	err = json.Unmarshal(respBytes, &resp)
+	if err != nil {
+		t.Fatalf("cannot unmarshal bytes: %s", respBytes)
+	}
+
+	if !strings.Contains(resp["error"], brokenHTTPRequest) {
+		t.Fatalf("Expected error to contain `%s`; error is `%s`", brokenHTTPResponse, resp["error"])
+	}
+}
+
+// Test that if a non json request is received that is greater than 8KB, that
+// we only put the first 8KB error, as to not flood memory with huge errors.
+func TestNonJSONRequestBodyIsLessThan8KB(t *testing.T) {
+	// Put a 16kb of bad data into the body
+	badBody := make([]byte, 8196*2)
+	for i := 0; i < 8196*2; i++ {
+		badBody = append(badBody, byte(i%256))
+	}
+	req, err := http.NewRequest("GET", httpAddr+"/error/non/json", bytes.NewReader(badBody))
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "cannot construct http request"))
+	}
+
+	respBytes, err := testHTTPRequest(req)
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "cannot make http request"))
+	}
+
+	var resp map[string]string
+	err = json.Unmarshal(respBytes, &resp)
+	if err != nil {
+		t.Fatalf("cannot unmarshal bytes: %s", respBytes)
+	}
+
+	l := len(resp["error"])
+	// Add 200 for padding for the actual error message in addition to the response body
+	if l > 8196+200 {
+		t.Fatalf("Expected error to be less than 8KB with a little padding, actual %d", l)
+	}
+	t.Log("Non JSON request length", l)
+}
+
+func TestResponseContentType(t *testing.T) {
+	req, err := http.NewRequest("GET", httpAddr+"/content/type", strings.NewReader(""))
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "cannot construct http request"))
+	}
+
+	client := &http.Client{}
+	httpResp, err := client.Do(req)
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "cannot make end http request"))
```

Comment:
Is this message supposed to say "cannot make end http request"? If so, could you elaborate on what it means? : )